reframe/core/meta.py

```diff
@@ -335,13 +335,12 @@ def __call__(cls, *args, **kwargs):
 
         # Intercept constructor arguments
         _rfm_use_params = kwargs.pop('_rfm_use_params', False)
-
+        _rfm_external_vals = kwargs.pop('_rfm_external_vals', {})
         obj = cls.__new__(cls, *args, **kwargs)
 
         # Insert the var & param spaces
-        cls._rfm_var_space.inject(obj, cls)
+        cls._rfm_var_space.inject(obj, cls, _rfm_external_vals)
```

Comment:
If we leave it to the loader to do just the `setattr` as mentioned above, there is no need for any of these changes in the metaclass and in the variables.